### PR TITLE
QMC tutorial edits

### PR DESCRIPTION
**Pull request by Atomyka**
Edits to formatting of QMC tutorial.

### CODE REVIEW

File: examples/7.QMC/main.jl [hunk 1]
```diff
@@ -44,37 +48,33 @@ atoms = generate_sites(ChainLattice(), nsites, scale = 5.72)
 
 Ω = 2π * 4
 Δ = 2π * 10
-h = rydberg_h(atoms; Δ, Ω)
+h = rydberg_h(atoms; Δ, Ω);
```
Comment (Roger-luo) [inline examples/7.QMC/main.jl:51]: why stop the printing here? It should give users a basic idea what Hamiltonian they are working on?

**Review by Roger-luo — DISMISSED** [review]
need to resolve merge conflicts before going further

File: examples/7.QMC/main.jl [hunk 2]
```diff
@@ -149,57 +149,43 @@ end
 
 using Plots: bar
     
-bar(occs_QMC, label="")
+bar = bar(occs_QMC, label="");
```
[the remark this answers is not in the document]
Comment (Atomyka) [reply examples/7.QMC/main.jl:152]: Ok

File: examples/7.QMC/main.jl (hunk 1 above)
Comment (Atomyka) [reply examples/7.QMC/main.jl:51]: Sure

**Comment by Atomyka** [comment]
Ready now, @Roger-luo ? :)

**Comment by weinbe58** [comment]
@Atomyka Can you copy over the changes I had on my branch [here](https://github.com/QuEraComputing/Bloqade.jl/blob/phil/QMC-example-fixes/examples/7.QMC/main.jl)

Its mostly equation fixes but also there was an issue with your analysis and defining the staggered magnetization.

**Comment by Atomyka** [comment]
> @Atomyka Can you copy over the changes I had on my branch [here](https://github.com/QuEraComputing/Bloqade.jl/blob/phil/QMC-example-fixes/examples/7.QMC/main.jl)
> 
> Its mostly equation fixes but also there was an issue with your analysis and defining the staggered magnetization.

@weinbe58 Done!

(Including the definition of the staggered mag you used. Previously I was using the definition Bloqade uses in the Adiabatic Evolution tutorial.)

**Review by johnzl-777 — CHANGES_REQUESTED** [review]
Did a local build of the example on my end and found some minor issues that cause it to crash. The following suggestions should fix it!

File: examples/7.QMC/main.jl [hunk 3]
```diff
@@ -149,88 +148,73 @@ end
...
-        BE = LogBinner(energy.(ns))
-        τ_energy = tau(BE)
-        ratio = 2 * τ_energy + 1
-        energy_binned = measurement(mean(BE), std_error(BE)*sqrt(ratio)) 
-        append!(energy_QMC, energy_binned)
+    for i in 1:MCS # Monte Carlo Steps
+        ns[i] = mc_step_beta!(rng, ts_ii, h_ii_qmc, β, d_ii, eq=false)
     end
 
+    # Binning analysis
```
[the remark this answers is not in the document]
Comment (Atomyka) [reply examples/7.QMC/main.jl:207]: Rearranged code to make it work!

File: examples/7.QMC/main.jl [hunk 4]
```diff
@@ -149,88 +148,73 @@ end
 
 using Plots: bar
     
-bar(occs_QMC, label="")
+bar = bar(densities_QMC, label="")
```
[the remark this answers is not in the document]
Comment (Atomyka) [reply examples/7.QMC/main.jl:151]: Makes sense!

**Comment by Atomyka** [comment]
Lmk if there is more!

**Review by johnzl-777 — APPROVED** [review]

File: examples/7.QMC/main.jl [hunk 5]
```diff
@@ -149,88 +148,72 @@ end
...
-        τ_energy = tau(BE)
-        ratio = 2 * τ_energy + 1
-        energy_binned = measurement(mean(BE), std_error(BE)*sqrt(ratio)) 
-        append!(energy_QMC, energy_binned)
+    for i in 1:MCS # Monte Carlo Steps
+        ns[i] = mc_step_beta!(rng, ts_ii, h_ii_qmc, β, d_ii, eq=false)
     end
 
+    BE = LogBinner(energy.(ns)) # Binning analysis 
+    energy(x) = -x / β + h_ii_qmc.energy_shift  # The energy shift here ensures that all matrix elements are non-negative. See Merali et al for details.
```
[the remark this answers is not in the document]
Comment (Atomyka) [reply examples/7.QMC/main.jl:207]: A yes, it slipped down. Edited! @johnzl-777

[the remark this answers is not in the document]
Comment (johnzl-777) [reply examples/7.QMC/main.jl:207]: Thanks 🤠 !

**Review by johnzl-777 — APPROVED** [review]

**Pull request merged**